flink-runtime/src/main/java/org/apache/flink/runtime/security/modules/HadoopModule.java

```diff
@@ -69,71 +60,34 @@ public void install() throws SecurityInstallException {
         UserGroupInformation loginUser;
 
         try {
-            if (UserGroupInformation.isSecurityEnabled()
-                    && !StringUtils.isBlank(securityConfig.getKeytab())
-                    && !StringUtils.isBlank(securityConfig.getPrincipal())) {
-                String keytabPath = (new File(securityConfig.getKeytab())).getAbsolutePath();
-
-                UserGroupInformation.loginUserFromKeytab(securityConfig.getPrincipal(), keytabPath);
```

Comment:
@steveloughran since you have quite some XP w/ UGI a quick question:
AFAIK `UserGroupInformation.loginUserFromKeytab` does the same just like
```
ugi = UserGroupInformation.loginUserFromKeytabAndReturnUGI
UserGroupInformation.setLoginUser(ugi);
```
right?

Reply:
yes, plus sets up a renewer thread to do relogin on a regular basis.